common/src/main/java/org/conscrypt/Conscrypt.java

```diff
@@ -59,6 +60,13 @@ public static void checkAvailability() {
     }
 
     /**
+     * Indicates whether the given object is a Conscrypt {@link Provider}.
+     */
+    public static boolean isConscrypt(Provider provider) {
+        return provider instanceof OpenSSLProvider;
```

Comment:
If we want to use this implementation, we should be more explicit about exactly what "is a Conscrypt Provider" means, namely that it only returns true if it's the provider that came from *this distribution* of Conscrypt.  This matters on Android because you can regularly have up to three different instances of Conscrypt in your classpath, and this will only return true for one of them.  In particular, it won't return true for Conscrypt.isConscrypt(Security.getProvider("AndroidOpenSSL")).

Reply:
@flooey I've updated the comments to reflect this.